src/app/pages/products-list/card/card.component.ts

```diff
@@ -1,8 +1,21 @@
-import {Component} from '@angular/core';
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+import {Product} from '../../../shared/products/product.interface';
 
 @Component({
     selector: 'app-card',
     templateUrl: './card.component.html',
     styleUrls: ['./card.component.css'],
 })
-export class CardComponent {}
+export class CardComponent {
+    @Input() product: Product | undefined;
+    @Output() buyClicked = new EventEmitter();
+
+    get imageUrl() {
```

Comment:
Для геттеров лучше писать явно возвращаемый ими тип

```suggestion
    get imageUrl(): string | undefined {
```

Reply:
запомню, спасибо!